extensions/silero_tts/script.py

```diff
@@ -1,349 +1,356 @@
-            voice = gr.Dropdown(value=params['speaker'], choices=languages[params['language']]["voices"], label='TTS voice')
-        with gr.Row():
-            v_pitch = gr.Dropdown(value=params['voice_pitch'], choices=voice_pitches, label='Voice pitch')
-            v_speed = gr.Dropdown(value=params['voice_speed'], choices=voice_speeds, label='Voice speed')
-
-        with gr.Row():
-            preview_text = gr.Text(show_label=False, placeholder="Preview text", elem_id="silero_preview_text")
-            preview_play = gr.Button("Preview")
-            preview_audio = gr.HTML(visible=False)
-
-        with gr.Row():
-            convert = gr.Button('Permanently replace audios with the message texts')
-            convert_cancel = gr.Button('Cancel', visible=False)
-            convert_confirm = gr.Button('Confirm (cannot be undone)', variant="stop", visible=False)
-
-    # Convert history with confirmation
-    convert_arr = [convert_confirm, convert, convert_cancel]
-    convert.click(lambda: [gr.update(visible=True), gr.update(visible=False), gr.update(visible=True)], None, convert_arr)
-    convert_confirm.click(
-        lambda: [gr.update(visible=False), gr.update(visible=True), gr.update(visible=False)], None, convert_arr).then(
-        remove_tts_from_history, gradio('history'), gradio('history')).then(
-        chat.save_history, gradio('history', 'unique_id', 'character_menu', 'mode'), None).then(
-        chat.redraw_html, gradio(ui_chat.reload_arr), gradio('display'))
-
-    convert_cancel.click(lambda: [gr.update(visible=False), gr.update(visible=True), gr.update(visible=False)], None, convert_arr)
-
-    # Toggle message text in history
-    show_text.change(
-        lambda x: params.update({"show_text": x}), show_text, None).then(
-        toggle_text_in_history, gradio('history'), gradio('history')).then(
-        chat.save_history, gradio('history', 'unique_id', 'character_menu', 'mode'), None).then(
-        chat.redraw_html, gradio(ui_chat.reload_arr), gradio('display'))
-
-    # Event functions to update the parameters in the backend
-    activate.change(lambda x: params.update({"activate": x}), activate, None)
-    autoplay.change(lambda x: params.update({"autoplay": x}), autoplay, None)
-    language.change(language_change, language, voice, show_progress=False)
-    voice.change(lambda x: params.update({"speaker": x}), voice, None)
-    v_pitch.change(lambda x: params.update({"voice_pitch": x}), v_pitch, None)
-    v_speed.change(lambda x: params.update({"voice_speed": x}), v_speed, None)
-
-    # Play preview
-    preview_text.submit(voice_preview, preview_text, preview_audio)
-    preview_play.click(voice_preview, preview_text, preview_audio)
+import html
+import json
+import random
+import time
+from pathlib import Path
+
+import gradio as gr
+import torch
+
+from extensions.silero_tts import tts_preprocessor
+from extensions.elevenlabs_tts import script as elevenlabs

```

Comment:
wait... so you are using the other extension inside this one? I don't know if this is the best approach. I would rather have access to the elevenlabs package directly here.
But either way, you do not commit that other extension here.